File: client/src/app/domains/entity/Report.ts
import { ReportReason } from '../enums/ReportReason';
import Comment from './Comment';
import { Post } from './Post';
import User from './User';

export default interface Report {
  id: number;
  reportReason: ReportReason;
  timestamp: String;
  reportedPost: Post | null;
  byUser: User;
  reportedComment: Comment | null;
}

File: yoloq/src/main/java/com/example/yoloq/models/Report.java
package com.example.yoloq.models;


import com.example.yoloq.enums.ReportReason;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table
@Getter
@Setter
@NoArgsConstructor
public class Report {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false)
    private ReportReason reportReason;

    @Column(nullable = false)
    private LocalDateTime timestamp;

    @Column(nullable = false)
    private boolean isDeleted;

    @ManyToOne
    private Comment reportedComment;

    @ManyToOne
    private Post reportedPost;

    @ManyToOne
    private User byUser;

}
